### ISSUE TITLE
Error by density-based clustering

### ISSUE THREAD

**Issue by JoseGuzman** [spyking-circus/spyking-circus#169]
I wanted to test a ~ 2 minutes recording where I control the spikes to perform ground-truth experiments. Unfortunately, clustering returns an error.

What is an appropriate recording length to perform a clustering? Using the latest version (0.8.5) with the old param file with same_elec in [clustering] adjusted to 0.2

```bash
Running density-based clustering...
  0%|                                            |[00:00<?, ?it/s]Traceback (most recent call last):
 
  File "/home/segundo.martinez/miniconda3/envs/sorting/bin/spyking-circus-subtask", line 11, in <module>
    load_entry_point('spyking-circus', 'console_scripts', 'spyking-circus-subtask')()
  File "/home/segundo.martinez/git/spyking-circus/build/lib/circus/scripts/subtask.py", line 43, in main
    circus.launch(task, filename, nb_cpu, nb_gpu, use_gpu)
  File "/home/segundo.martinez/git/spyking-circus/build/lib/circus/__init__.py", line 21, in launch
    module.main(params, nb_cpu, nb_gpu, use_gpu)
  File "/home/segundo.martinez/git/spyking-circus/build/lib/circus/clustering.py", line 771, in main
    save=save, alpha=sensitivity
  File "/home/segundo.martinez/git/spyking-circus/build/lib/circus/shared/plot.py", line 198, in view_clusters
    results = model.fit()
  File "/home/segundo.martinez/miniconda3/envs/sorting/lib/python3.6/site-packages/statsmodels/robust/robust_linear_model.py", line 278, in fit
    check_weights=True).fit()
  File "/home/segundo.martinez/miniconda3/envs/sorting/lib/python3.6/site-packages/statsmodels/regression/_tools.py", line 48, in __init__
    raise ValueError(self.msg.format('weights'))
ValueError: NaN, inf or invalid value detected in weights, estimation infeasible.
`

**Comment by yger** [comment]
0.8.5 should not accept sim_same_elec anymore. It has been renamed
merging_param and you should have warnings. Isn't it the case (behaviour
will be the same) ?

Regarding the error, this is strange, since with two minutes you should
have enough spikes.. If the file is not too big, can you send it to me?

**Comment by yger** [comment]
I just tried with your dataset, and while I found an error (now patched, and happening because of plots with low number of spikes), I can not reproduce your error with 0.8.5. I also had to change the parameter file, putting
merging_method = distance
merging_param = 0.2 # local merge (max 1) means merge same electrode 
instead of 
sim_same_elec = 0.2
(this is equivalent)

**Comment by JoseGuzman** [comment]
It's kind of strange, it only happens when I use  24 CPUs. If I put 23 or less it works nicely.

> `spyking-circus continuous.dat -m whitening,clustering,fitting,converting -c24`

```bash
-------------------------  Informations  -------------------------
| Number of recorded channels : 67
| Number of analyzed channels : 16
| File format                 : RAW_BINARY
| Data type                   : int16
| Sampling rate               : 30 kHz
| Duration of the recording   : 2 min 7 s 667 ms
| Width of the templates      : 2 ms
| Spatial radius considered   : 100 um
| Threshold crossing          : negative
------------------------------------------------------------------
Analyzing data to get whitening matrices and thresholds...
Found 4.23527s without spikes for whitening matrices...
Because of whitening, need to recompute the thresholds...
Searching spikes to construct the PCA basis...
100%|████████████████████████████████████|[00:00<00:00,  2.01it/s]
Found 81 isolated waveforms over 12792 requested
-------------------------  Informations  -------------------------
| A basis with 5 dimensions has been built
------------------------------------------------------------------
Searching random spikes to sample amplitudes...
100%|████████████████████████████████████|[00:08<00:00,  2.93it/s]
Found 195 spikes over 127992 requested
Estimating amplitudes distributions...
Smart Search of good isolated spikes for the clustering (1/3)...
100%|████████████████████████████████████|[00:08<00:00,  4.25it/s]
Found 172 isolated spikes over 127992 requested (4 rejected)
Computing density estimations...
Searching random spikes to refine the clustering (2/3)...
100%|████████████████████████████████████|[00:00<00:00,  2.61it/s]
Found 28 spikes over 127992 requested
Refining density estimations...
Searching random spikes to refine the clustering (3/3)...
100%|████████████████████████████████████|[00:00<00:00,  1.77it/s]
Found 24 spikes over 127992 requested
Refining density estimations...
Running density-based clustering...
  0%|                                            |[00:00<?, ?it/s]Traceback (most recent call last):
  File "/home/segundo.martinez/miniconda3/envs/sorting/bin/spyking-circus-subtask", line 11, in <module>
    load_entry_point('spyking-circus', 'console_scripts', 'spyking-circus-subtask')()
  File "/home/segundo.martinez/git/spyking-circus/build/lib/circus/scripts/subtask.py", line 43, in main
    circus.launch(task, filename, nb_cpu, nb_gpu, use_gpu)
  File "/home/segundo.martinez/git/spyking-circus/build/lib/circus/__init__.py", line 21, in launch
    module.main(params, nb_cpu, nb_gpu, use_gpu)
  File "/home/segundo.martinez/git/spyking-circus/build/lib/circus/clustering.py", line 771, in main
    save=save, alpha=sensitivity
  File "/home/segundo.martinez/git/spyking-circus/build/lib/circus/shared/plot.py", line 199, in view_clusters
    results = model.fit()
  File "/home/segundo.martinez/miniconda3/envs/sorting/lib/python3.6/site-packages/statsmodels/robust/robust_linear_model.py", line 278, in fit
    check_weights=True).fit()
  File "/home/segundo.martinez/miniconda3/envs/sorting/lib/python3.6/site-packages/statsmodels/regression/_tools.py", line 48, in __init__
    raise ValueError(self.msg.format('weights'))
ValueError: NaN, inf or invalid value detected in weights, estimation infeasible.


```

 until 12 it works nicely. With the number of CPUs>12

**Comment by yger** [comment]
Weird, I do not see the problem here, even using 24 cores. But the problem is in the plot function, so it is vanishing when using make_plots = in the [clustering] section?

**Comment by JoseGuzman** [comment]
Solved in f4b0cad54bc86fb8b41e9857fe02becdbf20f74f
